include/bluetooth/iso.h

```diff
@@ -337,49 +349,44 @@ struct bt_iso_server {
  */
 int bt_iso_server_register(struct bt_iso_server *server);
 
-/** @brief Bind ISO channels
- *
- *  Bind ISO channels with existing ACL connections, Channel objects passed
- *  (over an address of it) shouldn't be instantiated in application as
- *  standalone.
+/** @brief Creates a CIG as a central
  *
- *  @param conns Array of ACL connection objects
- *  @param num_conns Number of connection objects
- *  @param chans Array of ISO Channel objects to be created
+ *  @param[in] param     The parameters used to create and enable the CIG.
+ *                       The QOS parameters are determined by the QOS field of
+ *                       the first BIS in the BIS list of this parameter.
+ *  @param[out] out_cig  Connected Isochronous Group object on success.
  *
  *  @return 0 in case of success or negative value in case of error.
  */
-int bt_iso_chan_bind(struct bt_conn **conns, uint8_t num_conns,
-		     struct bt_iso_chan **chans);
+int bt_iso_cig_create(const struct bt_iso_cig_create_param *param,
+		      struct bt_iso_cig **out_cig);
 
-/** @brief Unbind ISO channel
+/** @brief Terminates a CIG as a central
  *
- *  Unbind ISO channel from ACL connection, channel must be in BT_ISO_BOUND
- *  state.
+ *  Once called, all CIS that are part of the CIG will be disconnected.
+ *  @NOTE: We could also require that ALL CIS shall be disconnected first
  *
- *  Note: Channels which the ACL connection has been disconnected are unbind
- *  automatically.
- *
- *  @param chan Channel object.
+ *  @param cig    Pointer to the CIG structure.
  *
  *  @return 0 in case of success or negative value in case of error.
  */
-int bt_iso_chan_unbind(struct bt_iso_chan *chan);
+int bt_iso_cig_terminate(struct bt_iso_cig *cig);
 
-/** @brief Connect ISO channels
+/** @brief Connect an ISO channel on an ACL connection
+ *
+ *  Connect an ISO channel. The ISO channel must have been initialized in a CIG
+ *  first by calling bt_iso_cig_create.
  *
- *  Connect ISO channels, once the connection is completed each channel
- *  connected() callback will be called. If the connection is rejected
- *  disconnected() callback is called instead.
- *  Channel object passed (over an address of it) as second parameter shouldn't
- *  be instantiated in application as standalone.
+ *  Once the connection is completed the channel's connected() callback will be
+ *  called. If the connection is rejected disconnected() callback is called
+ *  instead.
  *
- *  @param chans Array of ISO channel objects
- *  @param num_chans Number of channel objects
+ *  @param cis  Pointer to the connected isochronous stream (CIS) channel
+ *  @param conn Connection object.
  *
  *  @return 0 in case of success or negative value in case of error.
  */
-int bt_iso_chan_connect(struct bt_iso_chan **chans, uint8_t num_chans);
+int bt_iso_chan_connect(struct bt_iso_chan *cis, struct bt_conn *conn);
```

Comment:
The alternative could be to have an API to link an ACL to a cis (one at the time) and then have the connect call take in a list of bt_iso_chans as the ole API. 
My issue with the old API was that this linking of ACL to CIS was tied into the bind operation which also created the group.

Reply:
> I believe this maps to the HCI_LE_Create_CIS HCI command at some point. 

Correct 

> I would preferer that it was able to take in an array of iso-channels to connect to an array of ACL connections.

The reason why I didn't do it that way, is that error handling is easier when it is done one by one; e.g. if the central attempts to set up 4 ISO channels but the peripheral only supports two, then this will fail after the 2nd channel has connected. If we do it with an array, then we need to either disconnect the 2 connect channels and return an error, or find a way to tell the caller that only 2/4 channels connected. If we do it one by one in the API then we can leave the error handling up to the upper layer/caller (some LE Audio GATT profiles may impose specific ways of handling this case), while also avoid a lot of complexity. 

I don't see a strong advantage from neither the stack or the caller to provide an array in this situation. 

> Maybe also mention the "side-effects" of this functions - it will also do setup_iso_data_path, hence require a series of pointers to be set prior to the call. If you have an overview of which, adding them here would be helpful.

That makes sense to do. 

> The alternative could be to have an API to link an ACL to a cis (one at the time) and then have the connect call take in a list of bt_iso_chans as the ole API.

I was personally never a fan of the linking/binding concept: Besides not being a core spec concept, it did not provide any functionality, and the result of linking/binding can very easily be provided by the `bt_iso_connect` function in my opinion. It feels like an unnecessary step.

Reply:
Good point about the error handling, I guess in time all controllers will support multiple calls to LE_Create_CIS - then I agree.

Reply:
@Casper-Bonde-Bose I had actually remembered the create_cis took a single ISO and ACL; since the HCI command works that way, I suppose it would work with multiple ISO and single ACL (unlike the current with multiple ISO and multiple ACL). I guess it all depends on whether you'd ever want 2/4 connections instead of either none or all.